tools/check_format.py

```diff
@@ -171,56 +172,84 @@ def checkFilePath(file_path):
 def executeCommand(command, error_message, file_path,
         regex=re.compile(r"^(\d+)[a|c|d]?\d*(?:,\d+[a|c|d]?\d*)?$")):
   try:
-    subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT)
+    output = subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT).strip()
+    if output:
+      return output.split("\n")
+    return []
   except subprocess.CalledProcessError as e:
       if (e.returncode != 0 and e.returncode != 1):
-          print "ERROR: something went wrong while executing: %s" % e.cmd
-          sys.exit(1)
-      # In case we can't find any line numbers, call printError at first.
-      printError("%s for file: %s" % (error_message, file_path))
+        return ["ERROR: something went wrong while executing: %s" % e.cmd]
+      # In case we can't find any line numbers, record an error message first.
+      error_messages = ["%s for file: %s" % (error_message, file_path)]
       for line in e.output.splitlines():
         for num in regex.findall(line):
-          printError("  %s:%s" % (file_path, num))
+          error_messages.append("  %s:%s" % (file_path, num))
+      return error_messages
+
+def fixHeaderOrder(file_path):
+  command = "%s --rewrite %s" % (HEADER_ORDER_PATH, file_path)
+  if os.system(command) != 0:
+    return ["header_order.py rewrite error: %s" % (file_path)]
+  return []
+
+def clangFormat(file_path):
+  command = "%s -i %s" % (CLANG_FORMAT_PATH, file_path)
+  if os.system(command) != 0:
+    return ["clang-format rewrite error: %s" % (file_path)]
+  return []
 
 def fixFilePath(file_path):
   if isBuildFile(file_path):
     if os.system("%s %s %s" % (ENVOY_BUILD_FIXER_PATH, file_path, file_path)) != 0:
-      printError("envoy_build_fixer rewrite failed for file: %s" % file_path)
+      return ["envoy_build_fixer rewrite failed for file: %s" % file_path]
     if os.system("%s -mode=fix %s" % (BUILDIFIER_PATH, file_path)) != 0:
-      printError("buildifier rewrite failed for file: %s" % file_path)
-    return
+      return ["buildifier rewrite failed for file: %s" % file_path]
+    return []
   fixFileContents(file_path)
   if file_path.endswith(DOCS_SUFFIX):
-    return
-  if not checkNamespace(file_path) or not checkProtobufExternalDepsBuild(
-      file_path) or not checkProtobufExternalDeps(file_path):
-    printError("This cannot be automatically corrected. Please fix by hand.")
-  command = "%s --rewrite %s" % (HEADER_ORDER_PATH, file_path)
-  if os.system(command) != 0:
-    printError("header_order.py rewrite error: %s" % (file_path))
-  command = "%s -i %s" % (CLANG_FORMAT_PATH, file_path)
-  if os.system(command) != 0:
-    printError("clang-format rewrite error: %s" % (file_path))
+    return []
 
+  error_messages = (checkNamespace(file_path) or
```

Comment:
No, this is some Python hackery. `error_messages` will be assigned the first of the return values that is non-empty (since `[]` is logically false) or `[]` if none of the calls return error messages. I changed the code to make this explicit for clarity.